clang/lib/Analysis/CFG.cpp

```diff
@@ -2085,9 +2085,24 @@ void CFGBuilder::prependAutomaticObjLifetimeWithTerminator(
   if (!BuildOpts.AddLifetime)
     return;
   BumpVectorContext &C = cfg->getBumpVectorContext();
+
+  // Note that B is the LocalScope iterator associated with the backpatched
+  // jump and E is the LocalScope iterator associated with the jump target.
+  //
+  // To go from B to E, one first goes up the scopes from B
+  // to PofB, then sideways in one scope from PofB to PofE and then down
+  // the scopes from PofE to E.
+  // The lifetime of all objects between B and PofE end.
+  //
+  // Get E's ancestor that has the same LocalScope as one of B's ancestors.
+  LocalScope::const_iterator PofE = E.shared_parent(B);
+  int Dist = B.distance(PofE);
+  if (Dist <= 0)
```

Comment:
Can the distance be less than 0?

Reply:
While `Dist` cannot be less than 0 if the `distance` function is called as intended, it is possible to provide arguments to it such that the distance between two LocalScope iterators is less than zero. For example, this can happen if the control comes here for a "forward" jumping `goto` statement (which should not ideally happen). Moreover, this kind of check is there in two other places too and they have used a `<=` check (see lines 1770 - 1777).